modules/sfp_badpackets.py

```diff
@@ -0,0 +1,246 @@
+    """Bad Packets:Footprint,Investigate,Passive:Search Engines:apikey:Obtain information for any malicious activities by target"""
+
+    opts = {
+        'api_key': '',
+        'checkaffiliates': True,
+        'subnetlookup': False,
+        'netblocklookup': True,
+        'maxnetblock': 24,
+        'maxsubnet': 24
+    }
+
+    # Option descriptions. Delete any options not applicable to this module.
+    optdescs = {
+        "api_key": "Bad packets API Key",
+        'checkaffiliates': "Check affiliates?",
+        'subnetlookup': "Look up all IPs on subnets which your target is a part of?",
+        'netblocklookup': "Look up all IPs on netblocks deemed to be owned by your target for possible blacklisted hosts on the same target subdomain/domain?",
+        'maxnetblock': "If looking up owned netblocks, the maximum netblock size to look up all IPs within (CIDR value, 24 = /24, 16 = /16, etc.)",
+        'maxsubnet': "If looking up subnets, the maximum subnet size to look up all the IPs within (CIDR value, 24 = /24, 16 = /16, etc.)"
+    }
+
+    results = None
+
+    errorState = False  
+    limit = 100
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+        
+
+    # What events is this module interested in for input
+    # For a list of all events, check sfdb.py.
+    def watchedEvents(self):
+        return ["IP_ADDRESS", "NETBLOCK_OWNER", "NETBLOCK_MEMBER",
+            "AFFILIATE_IPADDR"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["IP_ADDRESS", "MALICIOUS_IPADDR", "RAW_RIR_DATA",
+            "COUNTRY_NAME", "DESCRIPTION_CATEGORY", "DESCRIPTION_ABSTRACT"]
+
+    # Check whether the IP Address is malicious using Bad Packets API
+    # https://docs.badpackets.net/#operation/query
+    def queryIPAddress(self, qry, currentOffset):
+        params = {
+            'source_ip_address': qry.encode('raw_unicode_escape').decode("ascii", errors='replace'),
+            'limit': self.limit, 
+            'offset': currentOffset
+        }
+
+        headers = {
+            'Accept' : "application/json",
+            'Authorization' : "Token " + self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+          'https://api.badpackets.net/v1/query?' + urllib.parse.urlencode(params),
+          headers=headers,
+          timeout=15,
+          useragent=self.opts['_useragent']
+        )
+
+        return self.parseAPIResponse(res)
+        
+    # Parse API Response from Badpackets
+    def parseAPIResponse(self, res):
+        if res['content'] is None:
+            self.sf.info("No Bad Packets information found")
+            return None
+
+        # Error codes as mentioned in Bad Packets Documentation
+        if res['code'] == '400':
+            self.sf.error("Invalid IP Address", False)
+            self.errorState = True
```

Comment:
`errorState` should only be set when the API appears to be broken or the module is in some bad state. Basically it's a way to prevent any further querying with this module, so just a single incorrect query is not an appropriate place for it to be set.